src/preferences/upgrade.cpp

```diff
@@ -29,6 +30,45 @@ Upgrade::Upgrade()
 Upgrade::~Upgrade() {
 }
 
+namespace {
+// mapping to proactively move users to the new all-shader waveform types
+WaveformWidgetType::Type upgradeToAllShaders(WaveformWidgetType::Type waveformType) {
+    // TODO: convert `WaveformWidgetType::Type` to an enum class then shorten more `using enum ...`
+    using WWT = WaveformWidgetType;
+    switch (waveformType) {
+    case WWT::EmptyWaveform:
+    case WWT::SoftwareSimpleWaveform:
+    case WWT::SoftwareWaveform:
+    case WWT::GLVSyncTest:
+    case WWT::QtVSyncTest:
+    case WWT::AllShaderRGBWaveform:
+    case WWT::AllShaderLRRGBWaveform:
+    case WWT::AllShaderFilteredWaveform:
+    case WWT::AllShaderSimpleWaveform:
+    case WWT::AllShaderHSVWaveform:
+    case WWT::Count_WaveformwidgetType:
+        return waveformType;
+    case WWT::QtSimpleWaveform:
+    case WWT::GLSimpleWaveform:
+        return WaveformWidgetType::AllShaderSimpleWaveform;
+    case WWT::GLFilteredWaveform:
+    case WWT::GLSLFilteredWaveform:
+        return WaveformWidgetType::AllShaderFilteredWaveform;
+    case WWT::QtWaveform:
+    case WWT::HSVWaveform:
+    case WWT::RGBWaveform:
+    case WWT::GLRGBWaveform:
+    case WWT::GLSLRGBWaveform:
+    case WWT::QtRGBWaveform:
+    case WWT::GLSLRGBStackedWaveform:
+        return WaveformWidgetType::AllShaderHSVWaveform;
+    case WWT::QtHSVWaveform:
+        return WaveformWidgetType::AllShaderHSVWaveform;
```

Comment:
```suggestion
    case WWT::GLSLRGBStackedWaveform:
    case WWT::QtHSVWaveform:
        return WaveformWidgetType::AllShaderHSVWaveform;
```

Reply:
> Good catch, but your suggestion looks wrong. Do this instead:

Yes, I didn't check the upgrade behaviour in detail, just the proposed code, so yeah: migrating RGB -> RGB makes sense, of course.